core/src/main/scripts/import_data_validator/validateData.py

```diff
@@ -1422,13 +1427,139 @@ class SegValidator(Validator):
+                    continue
+                # 0 is the first base, and loc.end is not part of the segment
+                # 'chrom' has already been read, as column order is fixed
+                if parsed_coords[col_name] < 0 or (
+                        'chrom' in parsed_coords and
+                        parsed_coords[col_name] > self.chromosome_lengths[
+                                                      parsed_coords['chrom']]):
+                    self.logger.error(
+                        'Genomic position beyond end of chromosome '
+                        '(chr%(chr)s:0-%(end)s)',
+                        {'chr': parsed_coords['chrom'],
+                         'end': self.chromosome_lengths[
+                                    parsed_coords['chrom']]},
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+                    # not a valid coordinate usable in further validations
+                    del parsed_coords[col_name]
+            elif col_name == 'num.mark':
+                if not self.checkInt(value):
+                    self.logger.error(
+                        'Number of probes is not an integer',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+            elif col_name == 'seg.mean':
+                if not self.checkFloat(value):
+                    self.logger.error(
+                        'Mean segment copy number is not a number',
+                        extra={'line_number': self.line_number,
+                               'column_number': col_index + 1,
+                               'cause': value})
+            else:
+                raise RuntimeError('Could not validate column type: ' +
+                                   col_name)
+
+        if (
+                'loc.start' in parsed_coords and 'loc.end' in parsed_coords and
+                parsed_coords['loc.start'] >= parsed_coords['loc.end'] - 1):
+            self.logger.error(
+                'Start position of segment is not lower than end position',
+                extra={'line_number': self.line_number,
+                       'cause': '{}/{}'.format(parsed_coords['loc.start'],
+                                               parsed_coords['loc.end'])})
+
+        # TODO check for overlap and low genome coverage
+        # this could be implemented by sorting the segments for a patient
+        # by (chromosome and) start position and checking if the start position
+        # of each segment comes after the end position of the previous one,
+        # meanwhile adding up the number of (non-overlapping) bases covered on
+        # that chromosome in that patient.
+
+    @staticmethod
+    def load_chromosome_lengths(genome_build):
```

Comment:
let's move this method and all the other web-services to another python file (services.py ? ). So we separate the validation logic from this kind of code. Maybe this applies to other parts as well, but let's just start with these.